### DISCUSSION TITLE
Creating custom native modules for third-party service

### DISCUSSION THREAD

**Discussion by swamptoday** [expo/expo#24280] [Archived: Questions] [answered]
As I have been already told - Expo gives the possibilty to create custom native modules based on some third-party services. 
However, it was mentioned that it is supported only in the development builds, not Expo Go. 

Can someone please explain how am I supposed to test what I wrote? Because, I truly have no idea on how everything works with custom native modules. For example, I tried [Use third-party native libraries in Expo](https://docs.expo.dev/modules/third-party-library/) and when executing it with Expo Go I got these errors:
```
1. ERROR  Error: Cannot find native module 'ExpoRadialChart', js engine: hermes
2.  ERROR  Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes
```
How am I supposed to check then constantly if code that I am writing will work correctly in the devlopment build? (maybe the questoin sounds stupid, but I am trying my best to explain). Thanks everyone in advance!

I intend to use this module like every other expo module in my already existing app.

**Comment by aleqsio** [comment, marked as answer]
Hi @swamptoday,

When working on a native module, we recommend using `npx expo run:ios` when you're working on a macbook – this should build your app to your simulator or plugged in iPhone for rapid testing.

Same thing for Android (except you can also do it on linux or windows)

**Reply by swamptoday** [reply]
So I can dynamically check my code with native modules implementation, if I am working with simulators? Perhaps you can be a little be more detailed on Android simulator - can I also somehow work with Android simulator if I start working with XCode, for example?
